### PR TITLE
[MM-24947] refactor cluster discovery store to use squirrel

### LINKED ISSUE #14503: Refactor SQL queries in store/sqlstore/cluster_discovery_store.go to use the squirrel builder
Mattermost is replacing the SQL queries of the store layer to use the [squirrel builder](https://github.com/Masterminds/squirrel). This help wanted issue is to change the usages of raw SQL queries in the file `store/sqlstore/cluster*discovery*store.go`.

The expected way to implement this is to search in the `store/sqlstore/cluster*discovery*store.go` file for all the strings containing a SQL query and rewrite them with squirrel. To do so, we must obtain the query builder using the `getQueryBuilder()` method from the store struct, and use it to build the same query that we have in the string.

You can find more information on how to use squirrel in [the official package documentation](https://godoc.org/github.com/Masterminds/squirrel).

Example Pull Request: https://github.com/mattermost/mattermost-server/pull/14393

If you have questions about the ticket or you need any help, feel free to contact `miguel.delacruz` in https://community.mattermost.com/

---
If you're interested please comment here and come [join our "Contributors" community channel](https://community.mattermost.com/core/channels/tickets) on our daily build server, where you can discuss questions with community members and the Mattermost core team. For technical advice or questions, please [join our "Developers" community channel](https://community.mattermost.com/core/channels/developers).

New contributors please see our [Developer's Guide](https://developers.mattermost.com/contribute/getting-started/).

JIRA: https://mattermost.atlassian.net/browse/MM-24947

### PR DESCRIPTION
**Pull request by kadir96**
#### Summary
This pull request refactors the cluster discovery store to use Squirrel instead of raw sql queries.

#### Ticket Link
Closes #14503
Jira ticket: https://mattermost.atlassian.net/browse/MM-24947

### CODE REVIEW

**Comment by agnivade** [comment]
I have some other PRs pending review. Please feel free to assign somebody else from Platform.

**Comment by teresa-novoa** [comment]
@lindy65 friendly reminder to review this PR, thanks!

**Comment by lindy65** [comment]
Thanks for the reminder @teresa-novoa - I'm just waiting on some testing direction from @mgdelacroix :)

File: store/sqlstore/cluster_discovery_store.go [hunk 1]
```diff
@@ -94,67 +87,56 @@ func (s sqlClusterDiscoveryStore) Exists(ClusterDiscovery *model.ClusterDiscover
...
+	query := s.getQueryBuilder().
+		Update("ClusterDiscovery").
+		Set("LastPingAt", model.GetMillis()).
+		Where(sq.Eq{"Type": ClusterDiscovery.Type}).
+		Where(sq.Eq{"ClusterName": ClusterDiscovery.ClusterName}).
+		Where(sq.Eq{"Hostname": ClusterDiscovery.Hostname})
+
+	queryString, args, err := query.ToSql()
+	if err != nil {
+		return model.NewAppError("SqlClusterDiscoveryStore.SetLastPingAt", "store.sql.build_query.app_error", nil, err.Error(), http.StatusInternalServerError)
```
Comment (ethervoid) [inline store/sqlstore/cluster_discovery_store.go:119]: Error Id should be `store.sql_cluster_discovery.set_last_ping.app_error`

**Pull request merged**